### PR TITLE
Add MasterNodesNeedResizingSRE alert

### PR DESCRIPTION
**Pull request by jewzaam**
https://issues.redhat.com/browse/OSD-3215

SOP is already merged: https://github.com/openshift/ops-sop/blob/master/v4/alerts/MasterNodesNeedResizingSRE.md

### CODE REVIEW

File: deploy/sre-prometheus/100-control-plane-resizing.PrometheusRule.yaml [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+        annotations:
+          message: "The cluster's master instance has been undersized for 15 minutes and should be vertically scaled to support the existing workers.  See linked SOP for details.  Critical alert will be raised at 2 hours."
+      - alert: MasterNodesNeedResizingSRE
+        expr: sre:node_masters:need_resize > 0
+        for: 2h
+        labels:
+          severity: critical
+          namespace: openshift-monitoring
+        annotations:
+          message: "The cluster's master instance has been undersized for 2 hours and must be vertically scaled to support the existing workers.  See linked SOP for details."
```
Comment (rogbas) [inline deploy/sre-prometheus/100-control-plane-resizing.PrometheusRule.yaml:56]: Where are we linking an SOP for this? Is it still TBD?

File: deploy/sre-prometheus/100-control-plane-resizing.PrometheusRule.yaml [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+        labels:
+          severity: warning
+          namespace: openshift-monitoring
+        annotations:
+          message: "The cluster's master instance has been undersized for 15 minutes and should be vertically scaled to support the existing workers.  See linked SOP for details.  Critical alert will be raised at 2 hours."
+      - alert: MasterNodesNeedResizingSRE
+        expr: sre:node_masters:need_resize > 0
+        for: 2h
+        labels:
+          severity: critical
```
Comment (RiRa12621) [inline deploy/sre-prometheus/100-control-plane-resizing.PrometheusRule.yaml:53]: No critical should come without immediate action

**Review by RiRa12621 — CHANGES_REQUESTED** [review]

File: deploy/sre-prometheus/100-control-plane-resizing.PrometheusRule.yaml (hunk 1 above)
Comment (jewzaam) [reply deploy/sre-prometheus/100-control-plane-resizing.PrometheusRule.yaml:56]: Auto-generated based on alert name: https://github.com/openshift/ops-sop/blob/master/v4/alerts/MasterNodesNeedResizingSRE.md

File: deploy/sre-prometheus/100-control-plane-resizing.PrometheusRule.yaml (hunk 2 above)
Comment (jewzaam) [reply deploy/sre-prometheus/100-control-plane-resizing.PrometheusRule.yaml:53]: https://github.com/openshift/ops-sop/blob/master/v4/alerts/MasterNodesNeedResizingSRE.md 
Action: resize the masters.

**Comment by jharrington22** [comment]
Seems reasonable, Looks good to me.

File: deploy/sre-prometheus/100-control-plane-resizing.PrometheusRule.yaml (hunk 1 above)
Comment (jharrington22) [reply deploy/sre-prometheus/100-control-plane-resizing.PrometheusRule.yaml:56]: Non blocking comment but how does that auto gen work?

**Review by jharrington22 — APPROVED** [review]

File: deploy/sre-prometheus/100-control-plane-resizing.PrometheusRule.yaml (hunk 2 above)
Comment (RiRa12621) [reply deploy/sre-prometheus/100-control-plane-resizing.PrometheusRule.yaml:53]: So if this pages during weekend oncall, I go ahead and resize the master nodes?

**Pull request merged**